rfcs/2021-12-06-10298-kubernetes_logs-rewrite.md

```diff
@@ -0,0 +1,115 @@
+# RFC 10298 - 2021-12-06 - `kubernetes_logs` Rewrite
+
+Rewriting the `kubernetes_logs` source to leverage the "official" Rust client for Kubernetes.
+
+## Context
+
+- [`kubernetes_logs` error handling RFC](https://github.com/vectordotdev/vector/issues/7527)
+- [`kubernetes_logs` source fixes](https://github.com/vectordotdev/vector/issues/10016)
+
+## Scope
+
+### In scope
+
+- Replacing our in-house Kubernetes code with `kube` wherever possible
+
+### Out of scope
+
+- Changing the end-to-end functionality of the `kubernetes_logs` source
+- Changing the testing strategies we have around the `kubernetes_logs` source today
+
+## Pain
+
+- Difficulty troubleshooting and supporting this feature as maintainers
+- Unresolved bug reports from users related to our in-house Kubernetes code
```

Comment:
Can you list out the specific bug reports? That'll help gauge the value of this change.